cpp/src/gis/point_in_polygon.cu

```diff
@@ -0,0 +1,184 @@
+ 
+    for(gdf_index_type idx = start_idx; idx < point_size; idx += blockDim.x * gridDim.x)
+    {
+        T point_lat = point_lats[start_idx];
+        T point_lon = point_lons[start_idx];
+        gdf_size_type count = 0;
+ 
+        for(gdf_index_type poly_idx = 0; poly_idx < poly_size - 1; poly_idx++) 
+        {
+            if(poly_lons[poly_idx] <= point_lon && point_lon < poly_lons[poly_idx + 1])
+            {
+                if (orientation(poly_lons[poly_idx], poly_lats[poly_idx], poly_lons[poly_idx + 1], poly_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                {
+                    count++;
+                }
+            }
+            else if (point_lon <= poly_lons[poly_idx] && poly_lons[poly_idx + 1] < point_lon) 
+            {
+                if (orientation(poly_lons[poly_idx], poly_lats[poly_idx], poly_lons[poly_idx + 1], poly_lats[poly_idx + 1], point_lon, point_lat) > 0)
+                {
+                    count++;
+                }
+            }
+        }
+        if ((count > 0) && (count % 2 == 0)) point_is_in_polygon[start_idx] = 1;
+        else point_is_in_polygon[start_idx] = 0;
+    }
+}
+
+struct point_in_polygon_functor {
+    template <typename col_type>
+    static constexpr bool is_supported()
+    {
+        return std::is_arithmetic<col_type>::value ;
+    }
+ 
+    template <typename col_type, typename std::enable_if< is_supported<col_type>() >::type* = nullptr>
+    void operator()(gdf_column const & d_poly_lats, gdf_column const & d_poly_lons,
+                    gdf_column const & d_point_lats, gdf_column const & d_point_lons,
+                    gdf_column const & d_point_is_in_polygon)
+    {
+
+        gdf_size_type min_grid_size = 0, block_size = 0;
+        CUDA_TRY( cudaOccupancyMaxPotentialBlockSize(&min_grid_size, &block_size, point_in_polygon_kernel<col_type>) );
+
+        point_in_polygon_kernel<col_type> <<< min_grid_size, block_size >>> (static_cast<col_type*>(d_poly_lats.data), 
+                static_cast<col_type*>(d_poly_lons.data), static_cast<col_type*>(d_point_lats.data), static_cast<col_type*>(d_point_lons.data), 
+                d_poly_lats.size, d_point_lats.size, static_cast<int8_t*>(d_point_is_in_polygon.data) );
+    }
+
+    template <typename col_type, typename std::enable_if< !is_supported<col_type>() >::type* = nullptr>
+    void operator()(gdf_column const & d_poly_lats, gdf_column const & d_poly_lons,
+                    gdf_column const & d_point_lats, gdf_column const & d_point_lons,
+                    gdf_column const & d_point_is_in_polygon)
+    {
+        CUDF_FAIL("Non-arithmetic operation is not supported");
+    }
+};
+    
+}   //  namespace
+
+namespace cudf {
+namespace gis {
+
+gdf_column point_in_polygon(gdf_column const & polygon_lats, gdf_column const & polygon_lons,
+                            gdf_column const & point_lats, gdf_column const & point_lons)
+{
+    cudaStream_t stream;
```

Comment:
Instead of creating a stream, make this internal `cudf::gis::` function take a stream parameter defaulted to 0.

Reply:
I changed that. You can check whether is better now.